acinclude/krb5.m4

```diff
@@ -8,6 +8,51 @@
 dnl these checks must be performed in the same order as here defined,
 dnl and have mostly been lifted out of an inlined configure.ac.
 
+AC_DEFUN([SQUID_CHECK_SOLARIS_KRB5],[
+  # no pkg-config for solaris native Kerberos
+  AS_IF([test "$cross_compiling" = "no" -a "x$with_mit_krb5" != "xyes" -a "x$with_mit_krb5" != "xno"],[
+    AC_PATH_PROG(krb5_config,krb5-config,no,[$PATH:$with_mit_krb5/bin])
+  ])
+  AC_MSG_NOTICE([Use krb5-config to get CFLAGS and LIBS])
+  LIBMIT_KRB5_CFLAGS="-I/usr/include/kerberosv5 `$ac_krb5_config --cflags krb5 2>/dev/null`"
+  LIBMIT_KRB5_LIBS="`$ac_krb5_config --libs krb5 2>/dev/null`"
+  # Solaris 10 Update 11 patches the krb5-config tool to produce stderr messages on stdout.
+  SOLARIS_BROKEN_KRB5CONFIG_GSSAPI="`$ac_krb5_config --libs gssapi 2>/dev/null | grep "krb5-config"`"
+  AS_IF([test "x$SOLARIS_BROKEN_KRB5CONFIG_GSSAPI" = "x"],[
+    LIBMIT_KRB5_CFLAGS="$LIBMIT_KRB5_CFLAGS `$ac_krb5_config --cflags gssapi 2>/dev/null`"
+    LIBMIT_KRB5_LIBS="$LIBMIT_KRB5_LIBS `$ac_krb5_config --libs gssapi 2>/dev/null`"
+  ])
+  CPPFLAGS="$LIBMIT_KRB5_CFLAGS $CPPFLAGS"
+  LIBS="$LIBMIT_KRB5_LIBS $LIBS"
+  missing_required=no
+  AC_MSG_NOTICE([Try to find Kerberos libraries in given path])
+  AC_CHECK_LIB(krb5, [main], [LIBMIT_KRB5_LIBS="$LIBMIT_KRB5_LIBS -lkrb5"],[
+    AC_MSG_WARN([library 'krb5' is required for Solaris Kerberos])
+    missing_required=yes
+  ])
+  AC_CHECK_LIB(gss, [main], [LIBMIT_KRB5_LIBS="$LIBMIT_KRB5_LIBS -lgss"],[
+    AC_MSG_WARN([library 'gss' is required for Solaris Kerberos])
+    missing_required=yes
+  ])
+  AS_IF([test "x$missing_required" = "xyes"],[LIBMIT_KRB5_LIBS=""],[
+    LIBS="$LIBMIT_KRB5_LIBS $LIBS"
+    AC_DEFINE(USE_SOLARIS_KRB5,1,[Solaris Kerberos support is available])
+    SQUID_CHECK_KRB5_SOLARIS_BROKEN_KRB5_H
+    AS_IF([test "x$squid_cv_broken_krb5_h" = "xyes"],[
+      AC_DEFINE(HAVE_BROKEN_SOLARIS_KRB5_H, 1, [Define to 1 if Solaris krb5.h is broken for C++])
+      AC_MSG_WARN([You have a broken Solaris <krb5.h> system include.])
+      AC_MSG_WARN([Please see http://bugs.opensolaris.org/bugdatabase/view_bug.do?bug_id=6837512])
+      AC_MSG_WARN([If you need Kerberos support you will have to patch])
+      AC_MSG_WARN([your system. See contrib/solaris/solaris-krb5-include.patch])
+    ])
+    AC_CHECK_HEADERS(gssapi.h gssapi/gssapi.h gssapi/gssapi_krb5.h)
```

Comment:
Similar concern here: We already check gssapi.h elsewhere, under different conditions.

Reply:
Not to worry, autoconf uses result caching to reduce work when the cases are the same. And when they are different we do need to re-check with the different `$PATH` anyway.

Reply:
I am not worried about autoconf ability to set the value of a HAVE_FOO macro. I am worried about the code that may be assuming that a true value implies that certain preconditions (specific to that code) were met. With two independent checks, there are two sets of preconditions, so that assumption (if it exists in code) will be wrong.

On the other hand, if there are no such assumptions in our code, then we should be testing the corresponding headers unconditionally, like we already do with many other headers.

Either way, this code is problematic.